server/data_common/dataset_metadata.py

```diff
@@ -90,3 +90,36 @@ def get_dataset_metadata_for_explorer_location(dataset_explorer_location: str, a
         raise DatasetNotFoundError(f"Dataset location not found for {dataset_explorer_location}")
 
     return dataset_metadata
+
+
+def get_dataset_and_collection_metadata(dataset_explorer_location: str, app_config: AppConfig):
+    data_locator_base_url = app_config.server_config.get_data_locator_api_base_url()
+    web_base_url = app_config.server_config.get_web_base_url()
+
+    base_metadata = get_dataset_metadata_for_explorer_location(dataset_explorer_location, app_config)
```

Comment:
Can you use the cache here. Similar to how its done https://github.com/chanzuckerberg/single-cell-explorer/pull/64/files#diff-fbe8fa00f6e9f8f6796df09b5b4b9679567c96aa3882afe2c2bf10a3a6263d71L82